dig.go

```diff
@@ -314,6 +314,47 @@ type Container struct {
 
 	// invokerFn calls a function with arguments provided to Provide or Invoke.
 	invokerFn invokerFn
+
+	gh *graphHolder
+}
+
+// graphHolder represents the dependency graph for a Container. Specifically,
```

Comment:
nit: can we move graphHolder and graphNode to a graph.go. (There's already one, but that's for graph visualization and can be renamed to visualize.go.)